sdk/storage/azure-storage-files-shares/sample/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required (VERSION 3.13)
+
+target_sources(
+    azure-storage-sample
+    PRIVATE
+    file_share_getting_started.cpp
+)
+
+target_link_libraries(azure-storage-sample PUBLIC Azure::azure-storage-files-shares)
```

Comment:
This is odd, why isn't it Azure::Storage::Files::Shares?

Reply:
This is not what you think it is (NOT a C++ namespace), and this is not how it works, in my understanding - you are only allowed to have one "namespace".

Reply:
It is an CMake target ALIAS,  it can also be Azure::Storage::Files::Shares, but usually people use only one groouping label + the CMake target name. The `Azure` is the grouping label, and then there is the name of the CMake target.
When this ALIAS was created looks like:
add_library(azure::storage::files::shares ALIAS azure-storage-files-shares), so it should work using it like that

Reply:
But this is not the target name that VcPkg is going to produce. VcPkg is going to tell you
```
The package azure-storage-files-shares-cpp:x64-windows-static provides CMake targets:

    find_package(azure-storage-files-shares-cpp CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Azure::azure-storage-files-shares)
```

So then you have two different names you need to reference - one is alias `Azure::Storage::Files::Shares` which is good and valid when you build the entire SDK as a single project, and then you need an if() to provide `Azure::azure-storage-files-shares` as link target, when building as individual library (when VcPkg is getting built)

Reply:
You are getting
```
target_link_libraries(main PRIVATE Azure::azure-storage-files-shares)
```

Because when you created the ALIAS, you used `Azure::${TARGET_NAME}` which translates to Azure::azure-storage-files-shares.   The alias is expected be more like `Azure::Storage::Files::Shares`